udata/core/dataset/rdf.py

```diff
@@ -468,40 +469,29 @@ def title_from_rdf(rdf, url):
             return i18n._("Nameless resource")
 
 
-def access_rights_from_rdf(resource: RdfResource):
+def access_rights_from_rdf(resource: RdfResource) -> str | None:
     """
     Extract the access rights from a RdfResource
-    If the access rights is a URIRef, return it as a string.
-    If the access rights is a RdfResource, return the label of the RdfResource if any, or the identifier of the RdfResource.
+    Cardinality is 0..1.
     """
-    access_right_info = resource.value(DCT.accessRights)
-    if isinstance(access_right_info, (URIRef, Literal)):
-        return access_right_info.toPython()
-    elif isinstance(access_right_info, RdfResource):
-        rdfs_label = rdf_value(access_right_info, RDFS.label)
-        if rdfs_label:
-            return rdfs_label
-        else:
-            return access_right_info.identifier.toPython()
+    return rdf_value(resource, DCT.accessRights, parse_label=True)
 
 
 def licenses_from_rdf(resource: RdfResource) -> set[str]:
     """
     Extract licences from a RDF distribution.
     See `test_dataset_rdf.py > test_licenses_from_rdf` for examples of supported formats.
+    Cardinality is 0..n (although it should be 0..1 per the spec).
+    """
+    return rdf_values(resource, DCT.license, parse_label=True)
+
+
+def rights_from_rdf(resource: RdfResource) -> set[str]:
```

Comment:
This is SEMIC v3 only, so future proofing.
I'm fine with having it here now. Just mentioning it in case data.gouv side prefers to minimize changes and add it later.

Reply:
Agreed, future-proofing. But with the added benefit of being coherent with the historical behavior of the DCAT harvester : `License` has always been guessed from `dct:license` OR `dct:rights`.